### PR TITLE
feat: non-contiguous source range

### PR DESCRIPTION
**Pull request by sinui0**
This PR switches to using `RangeSet` for `Span` which prepares to support parsing spans from non-contiguous source data. This will be required to support parsing HTTP bodies with a `chunked` transfer encoding.

~~Let's hold off on merging this until we address https://github.com/tlsnotary/tlsn-utils/issues/14 and can pin the main branch.~~

### CODE REVIEW

**Comment by sinui0** [comment]
@themighty1 @th4s ready for review. We can transplant this to `tlsn-utils` after we get some stuff merged in

**Review by th4s — APPROVED** [review]
:+1:

File: src/http/span.rs [hunk 1]
```diff
@@ -208,6 +178,20 @@ pub(crate) fn parse_response_from_bytes(
     Ok(response)
 }
 
+/// Converts a `httparse::Header` to a `Header`.
+fn from_header(src: &Bytes, header: &httparse::Header) -> Header {
+    let name_range = get_span_range(src, header.name.as_bytes());
+    let value_range = get_span_range(src, header.value);
+    // Add 2 to the end of the range to account for the terminating \r\n.
+    let header_range = name_range.start..value_range.end + 2;
```
Comment (themighty1) [inline src/http/span.rs]: ` header_range` must also account for any optional trailing whitespaces that may exist between the value and CRLF.

**Review by themighty1 — CHANGES_REQUESTED** [review]

File: src/http/span.rs (hunk 1 above)
Comment (sinui0) [reply src/http/span.rs]: good catch, fixed.

**Review by themighty1 — APPROVED** [review]

**Pull request merged**